fs/span-manager/span_manager_test.go

```diff
@@ -224,16 +224,16 @@ func TestStateTransition(t *testing.T) {
 			if tc.isPrefetch {
 				err := m.ResolveSpan(tc.spanID, r)
 				if err != nil {
-					t.Fatalf("failed resolving the span for prefetch")
+					t.Fatalf("failed resolving the span for prefetch: %v", err)
 				}
 				state := s.state.Load().(spanState)
 				if state != fetched {
 					t.Fatalf("failed transitioning to Fetched state")
 				}
 			} else {
-				_, err := m.GetSpanContent(tc.spanID, 0, s.endUncompOffset, s.endUncompOffset-s.startUncompOffset)
+				_, err := m.GetSpanContent(tc.spanID, 0, s.endUncompOffset-s.startUncompOffset, s.endUncompOffset-s.startUncompOffset)
```

Comment:
This seems weird given then signature:
```
GetSpanContent(spanID soci.SpanId, offsetStart, offsetEnd, size soci.FileSize)
```

I guess it doesn't matter since we're just checking for errors, not that the returned content is correct, but this should at least pass a sniff test.

Reply:
This is actually the correct implementation, if we want to fetch the whole span. The previous implementation would cause a panic due to a segfault.

Reply:
Oh, I see. uncomp offset is relative to the full uncompressed buffer, but offsetStart/End in the definition is relative to the span, right?

Maybe creating a `span.Size()`  or `span.UncompressedSize()` function rather than having that logic all over would make it easier to read.

Reply:
Agreed.